File: solutions/cpp/0049.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        vector<vector<string>> ans;
        unordered_map<string, vector<string>> map;

        for (const string& str : strs) {
            string s = str;
            sort(s.begin(), s.end());
            map[s].push_back(str);
        }

        for (const auto& i : map) {
            vector<string> s = i.second;
            sort(s.begin(), s.end());
            ans.push_back(s);
        }

        return ans;
    }
};

File: solutions/python3/0049.py
from collections import defaultdict

class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        ans = []
        dict = defaultdict(list)

        for str in strs:
            s = ''.join(sorted(str))
            dict[s].append(str)

        for value in dict.values():
            ans.append(value)

        return ans